File: misc/demo/unity3d/xlsxtools/Assets/Scripts/Config/ItemItemConfigTable.cs
using System.Collections;
using System.Collections.Generic;
public class ItemItemConfig
{
	public	int	id;	//id
	public	string	name;	//名称
	public	string	desc;	//描述
	public	string	gain;	//获取途径
	public	int	type;	//类型
	public	int	overlap;	//叠加上限
	public	int	usetype;	//使用效果类型
	public	int[]	usevalue;	//使用效果值2
	public	int	use_times;	//使用次数限制
	public	int	cansell;	//是否可出售
	public	ItemItemSellConfig[]	sell;
	public	string	head;	//小图标
	public	string	model;	//模型
	public	int	star;	//星级
	public	int	linkid;	//关联id
}

public class ItemItemSellConfig
{
	public int	id;	//出售获得物品1
	public int	num;	//出售获得物品数量1
}

public class ItemItemConfigTable
{
	Dictionary<int, ItemItemConfig> m_configs = new Dictionary<int,ItemItemConfig>();
	public Dictionary<int, ItemItemConfig> configs
	{
		get {
			return m_configs;
		}
	}

	public ItemItemConfig GetConfigById(int cid)
	{
		if(m_configs.ContainsKey(cid)){
			return m_configs[cid];
		}
		return null;
	}

	public object ConfigProcess(string[] row)
	{
		if (row.Length < 19)
		{
			return null;
		}

		RowHelper rh = new RowHelper(row);
		ItemItemConfig rec = new ItemItemConfig();

		rec.id = CSVUtility.ToInt(rh.Read());	//id
		rec.name = rh.Read();	//名称
		rec.desc = rh.Read();	//描述
		rec.gain = rh.Read();	//获取途径
		rec.type = CSVUtility.ToInt(rh.Read());	//类型
		rec.overlap = CSVUtility.ToInt(rh.Read());	//叠加上限
		rec.usetype = CSVUtility.ToInt(rh.Read());	//使用效果类型
		rec.usevalue = new int[2];
		rec.usevalue[0] = CSVUtility.ToInt(rh.Read());	//使用效果值1
		rec.usevalue[1] = CSVUtility.ToInt(rh.Read());	//使用效果值2
		rec.use_times = CSVUtility.ToInt(rh.Read());	//使用次数限制
		rec.cansell = CSVUtility.ToInt(rh.Read());	//是否可出售
		rec.sell = new ItemItemSellConfig[2];
		for (int i = 0;i < 2;i ++) {
			rec.sell[i] = new ItemItemSellConfig();
		}
		rec.sell[0].id = CSVUtility.ToInt(rh.Read());	//出售获得物品1
		rec.sell[0].num = CSVUtility.ToInt(rh.Read());	//出售获得物品数量1
		rec.sell[1].id = CSVUtility.ToInt(rh.Read());	//出售获得物品2
		rec.sell[1].num = CSVUtility.ToInt(rh.Read());	//出售获得物品数量2
		rec.head = rh.Read();	//小图标
		rec.model = rh.Read();	//模型
		rec.star = CSVUtility.ToInt(rh.Read());	//星级
		rec.linkid = CSVUtility.ToInt(rh.Read());	//关联id
		return rec;
	}

	public void Load()
	{
		CSVReader reader = new CSVReader();
		reader.LoadText("Config/Item_Item.txt", 3);
		int rows = reader.GetRowCount();
		for (int r = 0; r < rows; ++r)
		{
			string[] row = reader.GetRow(r);
			ItemItemConfig ac = ConfigProcess(row) as ItemItemConfig;
			configs.Add(ac.id, ac);
		}
	}
}

File: misc/demo/cpp/xlsxtools/Config/ItemItemConfigTable.cpp
#include "stdafx.h"
#include "ItemItemConfigTable.h"
#include "CSVReader.h"


ItemItemConfigTable::ItemItemConfigTable()
{
}


ItemItemConfigTable::~ItemItemConfigTable()
{
}


map<int, ItemItemConfig> & ItemItemConfigTable::GetConfigs()
{
	return m_configs;
}

ItemItemConfig * ItemItemConfigTable::GetConfigById(int cid)
{
	if (m_configs.find(cid) != m_configs.end())
	{
		return &m_configs[cid];
	}
	return NULL;
}


bool ItemItemConfigTable::ConfigProcess(vector<string> fields, ItemItemConfig & rec)
{
	if (fields.size() < 19)

	{

		return false;

	}


	int idx = 0;


	rec.id = stoi(fields[idx++]);	//id

	rec.name = fields[idx++];	//名称

	rec.desc = fields[idx++];	//描述

	rec.gain = fields[idx++];	//获取途径

	rec.type = stoi(fields[idx++]);;	//类型

	rec.overlap = stoi(fields[idx++]);	//叠加上限

	rec.usetype = stoi(fields[idx++]);;	//使用效果类型

	rec.usevalue.push_back(stoi(fields[idx++]));	//使用效果值1

	rec.usevalue.push_back(stoi(fields[idx++]));	//使用效果值2

	rec.use_times = stoi(fields[idx++]);	//使用次数限制

	rec.cansell = stoi(fields[idx++]);	//是否可出售


	for (int i = 0; i < 2; i++) {

		rec.sell.push_back(ItemItemSellConfig());

	}

	rec.sell[0].id = stoi(fields[idx++]);	//出售获得物品1

	rec.sell[0].num = stoi(fields[idx++]);	//出售获得物品数量1

	rec.sell[1].id = stoi(fields[idx++]);	//出售获得物品2

	rec.sell[1].num = stoi(fields[idx++]);	//出售获得物品数量2

	rec.head = fields[idx++];	//小图标

	rec.model = fields[idx++];	//模型

	rec.star = stoi(fields[idx++]);	//星级

	rec.linkid = stoi(fields[idx++]);	//关联id

	return true;
}

void ItemItemConfigTable::Load()
{
	CSVReader reader;
	reader.LoadText("Config/Item_Item.txt", 3);
	int rows = reader.GetRowCount();
	for (int r = 0; r < rows; ++r)
	{
		vector<string> row = reader.GetRow(r);
		ItemItemConfig ac;
		ConfigProcess(row, ac);
		m_configs.insert(std::make_pair(ac.id, ac));
	}
}